lib/maths/analytics/unittest/CBoostedTreeTest.cc

```diff
@@ -792,6 +790,182 @@ BOOST_AUTO_TEST_CASE(testLowTrainFractionPerFold) {
+    fillDataFrame(rows, 0, cols, x, noise, target, *frame);
+
+    auto regression = maths::analytics::CBoostedTreeFactory::constructFromParameters(
+                          1, std::make_unique<maths::analytics::boosted_tree::CMse>())
+                          .endOfHoldoutRows(endOfHoldoutRows)
+                          .eta({0.02})
+                          .buildForTrain(*frame, cols - 1);
+
+    regression->train();
+    regression->predict();
+
+    auto newFrame = core::makeMainStorageDataFrame(cols).first;
+
+    fillDataFrame(rows, 0, cols, x, noise, target, *newFrame);
+
+    for (std::size_t i = 0; i < cols - 1; ++i) {
+        rng.generateUniformSamples(0.0, 10.0, extraTrainingRows, x[i]);
+    }
+    rng.generateNormalSamples(0.0, noiseVariance, extraTrainingRows, noise);
+
+    fillDataFrame(extraTrainingRows, 0, cols, x, noise, target, *newFrame);
+
+    core::CPackedBitVector oldTrainingRowMask{rows, true};
+    oldTrainingRowMask.extend(false, extraTrainingRows);
+    core::CPackedBitVector newTrainingRowMask{~oldTrainingRowMask};
+
+    double alpha{regression->hyperparameters().depthPenaltyMultiplier().value()};
+    double gamma{regression->hyperparameters().treeSizePenaltyMultiplier().value()};
+    double lambda{regression->hyperparameters().leafWeightPenaltyMultiplier().value()};
+    regression = maths::analytics::CBoostedTreeFactory::constructFromModel(std::move(regression))
+                     .endOfHoldoutRows(endOfHoldoutRows)
+                     .newTrainingRowMask(newTrainingRowMask)
+                     .depthPenaltyMultiplier({0.5 * alpha, 2.0 * alpha})
+                     .treeSizePenaltyMultiplier({0.5 * gamma, 2.0 * gamma})
+                     .leafWeightPenaltyMultiplier({0.5 * lambda, 2.0 * lambda})
+                     .buildForTrainIncremental(*newFrame, cols - 1);
+
+    regression->trainIncremental();
+    regression->predict();
+
+    // Test the minimum cross-validation error matches the error we compute for
+    // holdout set.
+
+    core::CPackedBitVector holdoutRowMask(endOfHoldoutRows, true);
+    holdoutRowMask.extend(false, rows - endOfHoldoutRows);
+    TMeanAccumulator expectedMse;
+    TMeanAccumulator expectedMsd;
+    newFrame->readRows(
+        1, 0, frame->numberRows(),
+        [&](const TRowItr& beginRows, const TRowItr& endRows) {
+            for (auto row = beginRows; row != endRows; ++row) {
+                double actual{(*row)[cols - 1]};
+                double prediction{regression->prediction(*row)[0]};
+                double previousPrediction{regression->previousPrediction(*row)[0]};
+                expectedMse.add(maths::common::CTools::pow2(actual - prediction));
+                expectedMsd.add(maths::common::CTools::pow2(prediction - previousPrediction));
+            }
+        },
+        &holdoutRowMask);
+    LOG_DEBUG(<< expectedMse << " " << expectedMsd);
+    LOG_DEBUG(<< maths::common::CBasicStatistics::mean(expectedMse) +
```

Comment:
What does this number mean?

Reply:
This was left over from debugging an issue. See [512fcde](https://github.com/elastic/ml-cpp/pull/2190/commits/512fcdee1660348dfdbda0f9ac35ac20fec042b2).